engine/script/src/script_msg.cpp

```diff
@@ -386,13 +386,13 @@ namespace dmScript
             {
                 if (lua_isstring(L, 2))
                 {
-                    const char* path = lua_tostring(L, 2);
+                    size_t path_length;
+                    const char* path = lua_tolstring(L, 2, &path_length);
                     if (lua_isnil(L, 1) || (lua_isstring(L, 1) && *lua_tostring(L, 1) == '\0'))
                     {
-                        size_t path_size = strlen(path);
-                        if (path_size > 0)
+                        if (path_length > 0)
                         {
-                            ResolvePath(L, path, path_size, url.m_Path);
+                            ResolvePath(L, path, path_length, url.m_Path);
```

Comment:
When talking about C strings, I'm used to size being the buffer size (i.e. including the space needed for terminating null), and length excluding the null character. When these terms are used interchangeably in the code base, it can be frustrating as the user needs to double-check what value is assigned to it. But it may not be as well-established terms as I've assumed.

In any case, you're right in that I shouldn't pollute the changeset with this kind of thing.